shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -1284,11 +1288,16 @@ routing message to a random node so that:
                                         "Produce response included error NOT_LEADER_OR_FOLLOWER and so cleared topic {:?}",
                                         topic_name,
                                     );
+                                    response_partition.current_leader =
```

Comment:
why do we set this twice?

Reply:
If I'm reading the code correctly, without this line `response_partition.current_leader` won't get cleared if `break;` happens?

Reply:
ah true.
I think we should pull the clearing logic added by this PR into a separate for loop.
This will make the logic easier to follow, as it will avoid entangling the separate processes of "modifying the message to hide the real cluster" and "reading the message to update shotover's internal metadata".

Reply:
Ok I can do it